File: libs/sdk-ruby/lib/daytona/common/resources.rb
# Copyright Daytona Platforms Inc.
# SPDX-License-Identifier: Apache-2.0

# frozen_string_literal: true

module Daytona
  class Resources
    # @return [Integer, nil] Number of CPU cores to allocate
    attr_reader :cpu

    # @return [Integer, nil] Amount of memory in GiB to allocate
    attr_reader :memory

    # @return [Integer, nil] Amount of disk space in GiB to allocate
    attr_reader :disk

    # @return [Integer, nil] Number of GPUs to allocate
    attr_reader :gpu

    # @return [String, Array<String>, nil] Preferred GPU type for the Sandbox
    attr_reader :gpu_type

    # @param cpu [Integer, nil] Number of CPU cores to allocate
    # @param memory [Integer, nil] Amount of memory in GiB to allocate
    # @param disk [Integer, nil] Amount of disk space in GiB to allocate
    # @param gpu [Integer, nil] Number of GPUs to allocate
    # @param gpu_type [String, Array<String>, nil] Preferred GPU type for the Sandbox
    def initialize(cpu: nil, memory: nil, disk: nil, gpu: nil, gpu_type: nil)
      @cpu = cpu
      @memory = memory
      @disk = disk
      @gpu = gpu
      @gpu_type = gpu_type
    end

    # @return [Hash] Hash representation of the resources
    def to_h = { cpu:, memory:, disk:, gpu:, gpu_type: }.compact
  end
end

File: libs/sdk-java/src/main/java/io/daytona/sdk/model/Resources.java
// Copyright Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

package io.daytona.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
/**
 * Resource allocation values for a Sandbox.
 */
public class Resources {
    @JsonProperty("cpu")
    private Integer cpu;
    @JsonProperty("gpu")
    private Integer gpu;
    @JsonProperty("memory")
    private Integer memory;
    @JsonProperty("disk")
    private Integer disk;

    /**
     * Returns CPU allocation in cores.
     *
     * @return CPU cores
     */
    public Integer getCpu() { return cpu; }

    /**
     * Sets CPU allocation in cores.
     *
     * @param cpu CPU cores
     */
    public void setCpu(Integer cpu) { this.cpu = cpu; }

    /**
     * Returns GPU allocation.
     *
     * @return GPU units
     */
    public Integer getGpu() { return gpu; }

    /**
     * Sets GPU allocation.
     *
     * @param gpu GPU units
     */
    public void setGpu(Integer gpu) { this.gpu = gpu; }

    /**
     * Returns memory allocation in GiB.
     *
     * @return memory allocation
     */
    public Integer getMemory() { return memory; }

    /**
     * Sets memory allocation in GiB.
     *
     * @param memory memory allocation
     */
    public void setMemory(Integer memory) { this.memory = memory; }

    /**
     * Returns disk allocation in GiB.
     *
     * @return disk allocation
     */
    public Integer getDisk() { return disk; }

    /**
     * Sets disk allocation in GiB.
     *
     * @param disk disk allocation
     */
    public void setDisk(Integer disk) { this.disk = disk; }
}
